app/scripts/modules/core/state.service.ts

```diff
@@ -1,8 +1,27 @@
-import {IStateParamsService, IStateService} from 'angular-ui-router';
+import { module, IRootScopeService } from 'angular';
+import { IStateParamsService, IStateService } from 'angular-ui-router';
+import { Subject } from 'rxjs/Subject';
 
-export let stateService: IStateService = undefined;
-export let stateParamsService: IStateParamsService = undefined;
+export class StateService {
```

Comment:
Newer version of ui-router exposes `$state` API as a class named `StateService`.   I think it would be wise to choose a different name for this.  

There is also this: https://github.com/ui-router/rx which is roughly the same thing as you wrote but only works with ui-router 1.0+